File: Data Structures & Algorithms/sort-an-array/submission-0.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        if len(nums) <= 1 : return nums

        middle :int = nums[len(nums)//2]

        return self.sortArray([num for num in nums if num < middle] )+ [num for num in nums if num == middle] + self.sortArray([num for num in nums if num > middle])
        

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {

        int i = 0 , j = nums.length - 1   ; 

        //Two Pointers from i -> [middle]  <- j  
         
        for( ; i < j ; )
        {
            int middle = (i+j)/2 ; 
            if(nums[middle] == target) return middle ; 

            if(nums[middle] < target) i = middle + 1  ;
            else  j = middle - 1 ;  
        }

        return  nums[(i+j)/2] == target ? (i+j)/2 : -1 ; 
        
    }
}
